drivers/rail/rail_drv.c

```diff
@@ -0,0 +1,798 @@
+    dev->lastRxPacketHandle = RAIL_RX_PACKET_HANDLE_INVALID;
+
+    // init other hal stuff
+    //    rail_internal_init_radio_hal(params);
+
+    //// init radio
+    //  RAIL_RfIdle();
+    //  dev->state = RAIL_TRANSCEIVER_STATE_IDLE;
+
+}
+
+#if (RAIL_PTI_ENABLED == 1)
+int _rail_PTI_init(rail_t *dev)
+{
+
+
+    // init gpio for output
+
+    RAIL_PtiConfig_t pti_config = RAIL_PTI_CONFIG;
+
+    RAIL_ConfigPti(dev->rhandle, &pti_config);
+
+    RAIL_EnablePti(dev->rhandle, true);
+
+
+    DEBUG("RADIO_PTI_Init done\n");
+
+    return 0;
+}
+#endif
+
+int _rail_PA_init(rail_t *dev)
+{
+
+    // Initialize the RAIL Tx power curves for all PAs on this chip
+    RAIL_TxPowerCurvesConfig_t tx_power_curves_config = {
+        curves24Hp,
+        curvesSg,
+        curves24Lp,
+        piecewiseSegments
+    };
+
+    RAIL_Status_t ret = RAIL_InitTxPowerCurves(&tx_power_curves_config);
+
+    if (ret != RAIL_STATUS_NO_ERROR) {
+        LOG_ERROR("Error init PA  (Tx power curves) for rail - error msg: %s\n", rail_error2str(ret));
+        return -1;
+    }
+
+    // Power config, depends on chip etc ...
+    // TODO multi freq, mult protocol
+    RAIL_TxPowerConfig_t tx_power_config = {
+    #if RAIL_RADIO_BAND == 2400
+        RAIL_TX_POWER_MODE_2P4_HP,      // 2.4GHZ HighPower, TODO low power?
+    #elif (RAIL_RADIO_BAND == 868) || (RAIL_RADIO_BAND == 915)
+        RAIL_TX_POWER_MODE_SUBGIG,
+    #endif
+        RAIL_RADIO_PA_VOLTAGE,      // voltage vPA for the DCDC connection
+        10
+    };
+
+    ret = RAIL_ConfigTxPower(dev->rhandle, &tx_power_config);
+
+    if (ret != RAIL_STATUS_NO_ERROR) {
+        LOG_ERROR("Error init PA  (config Tx power) for rail - error msg: %s\n", rail_error2str(ret));
+        return -1;
+    }
+
+    ret = RAIL_SetTxPowerDbm(dev->rhandle, ((RAIL_TxPower_t) RAIL_DEFAULT_TXPOWER) * 10);
+
+    if (ret != RAIL_STATUS_NO_ERROR) {
+        LOG_ERROR("Error init PA  (set tx power) for rail");
+        return -1;
+    }
+
+    DEBUG("RADIO_PA_Init done\n");
+
+    return 0;
+}
+
+
+int rail_init(rail_t *dev)
+{
+
+    RAIL_Status_t ret;
+    int r = 0;
+
+    netdev_ieee802154_t *netdev = (netdev_ieee802154_t *)dev;
+
+    // save ref for this driver, for global access
+    // TODO multible instances?
+    _rail_dev = dev;
+
+    // hack for receive frame
+    dev->recv_taken = false;
+
+
+    DEBUG("rail_init called\n");
+
+    dev->state = RAIL_TRANSCEIVER_STATE_UNINITIALIZED;
+
+    // start with long addr mode.
+    netdev->flags |= NETDEV_IEEE802154_SRC_MODE_LONG;
+
+
+    // get informations about the used raillib
+    // TODO check if driver is compatible?
+    RAIL_Version_t rail_version;
+    RAIL_GetVersion(&rail_version, true);
+
+    LOG_INFO("Using Silicon Labs RAIL Lib. Version %u.%u Rev: %u build: %u multiprotocol: %s \n",
+             rail_version.major, rail_version.minor, rail_version.rev, rail_version.build,
+             rail_version.multiprotocol ? "YES" : "NO");
```

Comment:
I'd advice not to mix `LOG_XXX` calls with `DEBUG` calls.

Reply:
OK, I though this could have info value, but it is probably just debug level